client/__tests__/util/stateManager/colorHelpers.test.ts

```diff
@@ -113,22 +114,27 @@ describe("categorical color helpers", () => {
     const data = obsDataframe.col("categoricalColumn").asArray();
     const cats = schemaClone.annotations.obsByName.categoricalColumn.categories;
     for (let i = 0; i < schemaClone.dataframe.nObs; i += 1) {
-      // @ts-expect-error ts-migrate(2722) FIXME: Cannot invoke an object which is possibly 'undefin... Remove this comment to see the full error message
-      expect(makeScale(ct.rgb[i])).toEqual(ct.scale(cats.indexOf(data[i])));
+      expect(makeScale(ct.rgb[i])).toEqual(
+        ct?.scale && ct?.scale(cats.indexOf(data[i]))
```

Comment:
why not:
```suggestion
        ct?.scale?.(cats.indexOf(data[i]))
```

Reply:
Thanks for flagging! `scale` is a function, so we need to check the function exists before invoking it!

Unfortunately we can't do `ct?.scale?(cats.indexOf(data[i]))`, because the compiler thinks it's ternary operation or something -.-

Reply:
It's ?.() not ?(), should work we use it in other places. https://github.com/chanzuckerberg/cellxgene/blob/eaae6df5e34682e5f96aadc3f1d7fc72ba342ca1/client/src/components/continuousLegend/index.js#L140

Reply:
OH! Totally didn't know that's available 😆 Thank you! Updated 👍